src/main/java/slimeknights/tconstruct/library/recipe/tinkerstation/ToolRecipes.java

```diff
@@ -0,0 +1,99 @@
+package slimeknights.tconstruct.library.recipe.tinkerstation;
+
+import lombok.Getter;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemStack;
+import slimeknights.tconstruct.library.materials.MaterialRegistry;
+import slimeknights.tconstruct.library.materials.definition.Material;
+import slimeknights.tconstruct.library.tools.definition.ToolDefinition;
+import slimeknights.tconstruct.library.tools.helper.ToolBuildHandler;
+import slimeknights.tconstruct.library.tools.helper.TooltipUtil;
+import slimeknights.tconstruct.library.tools.layout.LayoutSlot;
+import slimeknights.tconstruct.library.tools.layout.StationSlotLayout;
+import slimeknights.tconstruct.library.tools.layout.StationSlotLayoutLoader;
+import slimeknights.tconstruct.library.tools.part.IToolPart;
+import slimeknights.tconstruct.tools.item.ArmorSlotType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static slimeknights.tconstruct.library.tools.definition.module.ToolHooks.TOOL_PARTS;
+import static slimeknights.tconstruct.tools.TinkerTools.plateArmor;
+
+public class ToolRecipes {
+
+  @Getter
+  private final ToolDefinition definition;
+
+  private final StationSlotLayout slotLayout;
+
+  public ToolRecipes(ToolDefinition definition) {
+    this.definition = definition;
+    ResourceLocation id = switch(definition.getId().toString()) {
+      case "tconstruct:plate_helmet":
+      case "tconstruct:plate_chestplate":
+      case "tconstruct:plate_leggings":
+      case "tconstruct:plate_boots":
+        yield new ResourceLocation("tconstruct", "plate_armor");
+      default:
+        yield definition.getId();
+    };
+    this.slotLayout = StationSlotLayoutLoader.getInstance().get(id);
+  }
+
+  static List<Material> MATERIALS = MaterialRegistry.getMaterials()
+    .stream()
+    .map(material -> (Material) material)
+    .filter(material -> !material.isHidden())
+    .toList();
```

Comment:
This is not reload safe at all. An why are you doing an unchecked cast? `IMaterial` has everything you need. Just run this stream in the method that uses it.

Plus, I am certain we have a method to get all visible materials; we use that a ton. So just call the existing method.